### ISSUE TITLE
Assertion `getHandleCountDbg() < handlesLimit_ && "Too many handles allocated in GCScope"' failed.

### ISSUE THREAD

**Issue by Changochen** [facebook/hermes#279]
Git commit: `35da03ff3cd5c7c5391e809d8d1d99d8f227be0b`

CMD:`hermes poc.js`

POC:
```javascript
b = false
function c() {
  if (!b) {
    a = Array(3000)
    this[2] = a
  }
  b = true
}
JSON.parse("[1, 2, []]", c)
```

**Comment by dulinriley** [comment]
I've confirmed it's happening for me as well on master.

**Comment by dulinriley** [comment]
Fixed in 1be5a7bdccf47636ad93a0365be07c2b673be7d3, thanks for reporting this!
